HTTPServerProject/Interfaces.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.IO;
+
+namespace HTTPServerProject.Interfaces
+{
+    public interface INewStreamReader
+    {
+        string ReadLine();
+        void Close();
+    }
+    public abstract class NewStreamReader : INewStreamReader
+    {
+        Stream emptyStream = null!;
+        StreamReader reader = null!;
+        List<string> emptyArr = null!;
+        
+        public NewStreamReader(Stream stream = null!, List<string> arr = null!)
+        {
+            stream = emptyStream;
+            reader = new StreamReader(stream);
+            arr = emptyArr;
+        }
+
+        public abstract string ReadLine();
+
+        public abstract void Close();
+    }
+
+    public class MyStreamReader : NewStreamReader
+    {
+        Stream emptyStream = null!;
+        StreamReader reader = null!;
+        List<string> emptyArr = null!;
+        public MyStreamReader(Stream stream = null!, List<string> arr = null!)
+        {
+            stream = emptyStream;
+            reader = new StreamReader(stream);
+            arr = emptyArr;
+        }
+        public override string ReadLine()
+        {
+            string input = reader.ReadLine()!;
+            return input;
+        }
+
+        public override void Close()
+        {
+            reader.Close();
+        }
+    }
+
+    public class TestStreamReader : NewStreamReader
+    {
+        Stream emptyStream = null!;
```

Comment:
Doesn't all of this crash? My C# is rusty but if I remember right you're saying that "emptyStream" is not nullable (because the type is `Stream emptystream`  but then setting it to null anyway.

To get things to be nullable they need to have a question mark after the type name like `Stream?`.

Reply:
It hasn't crashed for me yet. Maybe they took away adding the `?` at the end to make variables null for version 6?

Reply:
I hope not - it's a beneficial feature, because if you don't want something to be nullable you don't have to specify.